autoload/codefmt/clangformat.vim

```diff
@@ -55,6 +55,36 @@ function! s:ClangFormatHasAtLeastVersion(minimum_version) abort
 endfunction
 
 
+" Inputs are 1-based (row, col) coordinates into lines.
+" Returns the corresponding zero-based offset into lines->join("\n")
+function! s:PositionToOffset(row, col, lines) abort
+  let l:offset = a:col - 1
+  if a:row > 1
+    for l:line in a:lines[0 : a:row - 2]
+      let l:offset += len(l:line) + 1
```

Comment:
The "+1" is for the newline, right? Could you comment one or two inline hints explaining the arbitrary-looking +N and -N parts?

Reply:
Done. The -2 is particularly awful.

(I wonder if there'd be enough usage of a maktaba#list#slice(begin, end) that returned a half-open range... `[begin, end)` - the `if` and extra `-1` to deal with closed ranges is horrible, and I don't know a better idiom)